tests/Doctrine/Tests/DBAL/Schema/SchemaTest.php

```diff
@@ -449,4 +451,48 @@ public function testVisitsNamespaceVisitor()
 
         $this->assertNull($schema->visit($visitor));
     }
+
+    /**
+     * https://github.com/doctrine/dbal/issues/2538
+     */
+    public function testFixGithub2538Collation()
+    {
+        $tableName = "public.foo";
+        $table = new Table($tableName);
+        $table->addColumn('test', Type::STRING, ['collation' => 'utf8_unicode_ci']);
+
+        $schema = new Schema(array($table));
+
+        $platform = new MySqlPlatform();
+
+        $this->assertSame(
+            ['CREATE TABLE public.foo (test VARCHAR(255) NOT NULL COLLATE utf8_unicode_ci) DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci ENGINE = InnoDB'],
+            $schema->toSql($platform),
+            $platform->getColumnDeclarationSQL('test')
+        );
+    }
+
+    /**
+     * https://github.com/doctrine/dbal/issues/2538
+     */
+    public function testFixGithub2538_autoincrement()
```

Comment:
Please use a method name describing the issue being covered. Also, use `@group #2538` for labeling it